x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/actions_log_table.tsx

```diff
@@ -97,30 +99,28 @@ export const ActionsLogTable = memo<ActionsLogTableProps>(
     );
 
     const toggleDetails = useCallback(
-      (item: ActionListApiResponse['data'][number]) => {
+      (actionId: ActionListApiResponse['data'][number]['id']) => {
         const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
-        if (itemIdToExpandedRowMapValues[item.id]) {
+        if (itemIdToExpandedRowMapValues[actionId]) {
           // close tray
-          delete itemIdToExpandedRowMapValues[item.id];
+          delete itemIdToExpandedRowMapValues[actionId];
         } else {
-          // expanded tray contents
-          itemIdToExpandedRowMapValues[item.id] = (
-            <ActionsLogExpandedTray action={item} data-test-subj={dataTestSubj} />
+          // find the action from the latest actions using id
+          const action = items.filter((item) => item.id === actionId)[0];
```

Comment:
This is kind of working. It shows updated data on the second toggle/open of the detail tray.